articles/tips-and-tricks-for-deep-learning/index.md

```diff
@@ -0,0 +1,252 @@
+3. It makes the network very difficult to train. 
+4. It decreases the training speed.      
+
+To avoid the problems mentioned above, we normalize their ages and mileage inputs to fit on the same scale. From our example, we can scale down our mileage and age data to a scale of between 0 to 1. Normalization reduces the wide ranges between the data points. However, normalization only solves part of the problem.
+
+From my previous [article](https://www.section.io/engineering-education/basics-of-convolution-neural-networks/), I discussed how training is achieved by updating the weights of the neural network iteratively through a process known as backpropagation. But within each iteration, these weights are updated simultaneously. This is a problem as it means that as the weights are updated, the weights of the previous layers are also updating simultaneously. This phenomenon is known as the internal covariate shift. It ends up slowing down the training as it requires the use of lower learning rates. Also, it makes it harder to train models as careful initialization of the parameters is needed. The internal covariate shift is the problem batch normalization aims to solve.
+
+Batch norm normalizes the information being passed between hidden layers to mitigate the internal covariate shift problem. This means that when information is passed from one hidden layer to another, the mean and standard deviation are calculated for every training mini-batch. This then helps us obtain a normalized output on each batch using the formula:
+
+![Batch Normalization formula](/engineering-education/tips-and-tricks-for-deep-learning/batch-normalization.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)*
+
+The formula can be simplified into two steps:
+- The first step subtracts the batch mean from every output value, dividing the result by the standard deviation. 
+The second step involves using the result obtained from the first step, multiplying it by the gamma hyperparameter, and adding it to the beta hyperparameter. 
+
+As a result, batch normalization allows the use of higher learning rates, increasing the training speed. It also eliminates the need for caring so much about the initialization of parameters.
+
+### Parameter tuning
+
+#### Weight Initialization
+
+The initial weight initialization of a neural network can affect how fast the model converges, how good the converged point is, and if the model converges at all.
+
+So, how do we choose the initial weight values?
+
+We could initialize all weights to zero. But is that a good idea?
+
+No. If we initialize all the parameters to the same value, we will get the same updates during training. This will make the model only learn the same features. That's not something we'd want in our model. Each neuron needs to learn something different to be useful.
```

Comment:
This will make the model only learn the same features -> This results in the model learning the same features.